WooCommerce/Classes/ViewRelated/Dashboard/Settings/SettingsViewController.swift

```diff
@@ -216,6 +225,31 @@ private extension SettingsViewController {
         present(alertController, animated: true)
     }
 
+    func switchStoreWasPressed() {
+        WooAnalytics.shared.track(.settingsSelectedStoreTapped)
+        StoresManager.shared.resetAuthentication()
+
+        let group = DispatchGroup()
+
+        group.enter()
+        let statsAction = StatsAction.resetStoredStats {
+            group.leave()
+        }
+        StoresManager.shared.dispatch(statsAction)
+
+        group.enter()
+        let orderAction = OrderAction.resetStoredOrders {
+            group.leave()
+        }
+        StoresManager.shared.dispatch(orderAction)
+
+        group.notify(queue: .main) {
+            let pickerVC = StorePickerViewController()
+            let loginNavController = LoginNavigationController(rootViewController: pickerVC)
+            self.present(loginNavController, animated: true, completion: nil)
```

Comment:
I would add `[weak self] in` to the `group.notify` block and call this as `self?.present(...`